apps/rush-lib/src/cli/actions/TabCompleteAction.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { BaseRushAction } from './BaseRushAction';
```

Comment:
There is a minor annoyance that the Rush process takes a second or two to spin up, because it has to `eval()` a lot of **node_modules** dependencies (e.g. `jz-zip`).  For commands like `rush tab-complete` that are really just doing simple string processing, we could significantly reduce this startup time by lazy-loading those dependencies.  

I'm interested to try out [import-lazy](https://github.com/sindresorhus/import-lazy) which is synchronous. It wraps imported modules with a proxy that will call `require()` automatically the first time the object is accessed.  We're already using it [in this file](https://github.com/microsoft/rushstack/blob/111cfda8a7f8371dcfb5b676c21f2de52fe1663e/stack/rush-stack-compiler-shared/src/shared/ToolPackages.js#L6).

A few judiciously placed calls to `import-lazy` might reduce our startup times considerably.

Reply:
Thanks @octogonz - let me see if I import-lazy improves spin up times.

Reply:
import-lazy + a few other optimizations here and there helped bring it down from ~2.2 seconds to ~0.7 seconds! Thanks @octogonz.